nf_core/modules/lint/main_nf.py

```diff
@@ -236,25 +236,30 @@ def check_process_section(self, lines, fix_version, progress_bar):
 
     # Check that process labels are correct
     correct_process_labels = ["process_single", "process_low", "process_medium", "process_high", "process_long"]
-    process_label = [l for l in lines if l.lstrip().startswith("label")]
-    if len(process_label) > 0:
-        try:
-            process_label = re.search("process_[A-Za-z]+", process_label[0]).group(0)
-        except AttributeError:
-            process_label = re.search("'([A-Za-z_-]+)'", process_label[0]).group(0)
-        finally:
-            if not process_label in correct_process_labels:
+    all_labels = [l for l in lines if l.lstrip().startswith("label ")]
+    bad_labels = []
+    good_labels = []
+    if len(all_labels) > 0:
+        for label in all_labels:
+            label = re.match("^label\s+([a-zA-Z0-9_-]+)", label).group(1)
```

Comment:
This will raise an exception if the regex does not match, e.g. if the label has forbidden characters. This was also the case before, but I guess it would be good to either check that `re.match(...)` is not `None` or put this into a `try`-block.

Reply:
Yes, I didn't change the functionality of a label which violates the regex... yet. I would like to perhaps add an extra check/warning indicating this (labels per nextflow docs can only be alphanumeric or underscore)

Reply:
Added a warning if a label is found which does not conform to the regex (I believe the regex is actually slightly more permissive than the true nextflow regulation for labels found [here](https://www.nextflow.io/docs/latest/process.html#label))